### PR TITLE
integrating istio as an addon on konvoy

### PR DESCRIPTION
**Pull request by GoelDeepak**

### CODE REVIEW

File: templates/istio.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+spec:
+  namespace: istio-system
+  requires:
+    - matchLabels:
+        kubeaddons.mesosphere.io/name: cert-manager
+  kubernetes:
+    minSupportedVersion: v1.15.0
+  cloudProvider:
+    - name: aws
+      enabled: false
```
Comment (shaneutt) [inline templates/istio.yaml]: Is it intentional for all the providers to be set to `enabled: false`?

Comment (GoelDeepak) [reply templates/istio.yaml]: yeah, the idea is to keep it `disabled` by default

File: templates/istio.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
+apiVersion: kubeaddons.mesosphere.io/v1beta1
+kind: ClusterAddon
+metadata:
+  name: istio
+  labels:
+    kubeaddons.mesosphere.io/name: istio
+  annotations:
+    catalog.kubeaddons.mesosphere.io/addon-revision: "1.3.3-1"
+    appversion.kubeaddons.mesosphere.io/istio: "1.3.3"
+    endpoint.kubeaddons.mesosphere.io/kiali: "/ops/portal/kiali"
```
Comment (s12chung) [inline templates/istio.yaml]: I'm curious, is this a new annotation type?

Comment (GoelDeepak) [reply templates/istio.yaml]: I copied it from other YAMLs. I am under impression that it is used by the UI team but I could very well be wrong here :).

**Pull request merged**